plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -13,27 +13,298 @@
+                .containsExactlyElementsOf(queryIds)
+                .size().isEqualTo(2);
+
+        // Multi-value insert
+        QueryId multiValueInsert = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("INSERT INTO %s VALUES (1, 2), (2, 1), (2, 2)", tableName))
+                .getQueryId();
+        queryIds.add(multiValueInsert);
+
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .containsExactlyElementsOf(queryIds)
+                .size().isEqualTo(3);
+
+        int snapshotIdCountBeforeUpgradeToV2 = getSnapshotsIdsByCreationOrder(tableName).size();
+
+        // Upgrade to Format v2 - this does not create a snapshot with a summary
+        QueryId alterTablePropertiesToFormatV2 = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("ALTER TABLE %s SET PROPERTIES format_version = 2", tableName))
+                .getQueryId();
+
+        // Ensure this query did not generate a snapshot and its associated query id is not in any summaries
+        assertThat(getSnapshotsIdsByCreationOrder(tableName))
+                .hasSize(snapshotIdCountBeforeUpgradeToV2);
+
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .doesNotContain(alterTablePropertiesToFormatV2);
+
+        // Delete one row in the file with 3 rows generated during multi-value insertion
+        QueryId rowLevelDelete = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("DELETE FROM %s WHERE a = 2 AND b = 1", tableName))
+                .getQueryId();
+        queryIds.add(rowLevelDelete);
+
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .containsExactlyElementsOf(queryIds)
+                .size().isEqualTo(4);
+
+        // Delete which generates a MOR file for the row (1, 2) and removes entirely the file for the row (1, 1).
+        // This generates two snapshots within the same transaction (both with the same query id)
+        QueryId rowLevelDeleteWhichAlsoDeletesWholeFile = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("DELETE FROM %s WHERE a = 1", tableName))
+                .getQueryId();
+        // Add twice - once for row level update and once for removal of file where all rows are now in delta files.
+        queryIds.add(rowLevelDeleteWhichAlsoDeletesWholeFile);
+        queryIds.add(rowLevelDeleteWhichAlsoDeletesWholeFile);
+
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .containsExactlyElementsOf(queryIds);
+
+        // Ensure there is a query id in each snapshot generated
+        assertThat(getSnapshotsIdsByCreationOrder(tableName))
+                .hasSameSizeAs(queryIds);
+    }
+
+    @Flaky(issue = ADD_QUERY_ID_TO_ICEBERG_SNAPSHOT_SUMMARY_ISSUE, match = ICEBERG_CATALOG)
```

Comment:
Why would we want to add a test that is known to be flaky?

Reply:
I don't think this test necessarily is flaky, but when I added it, something required me to add this annotation. I had assumed that maybe the `ALTER TABLE ... EXECUTE` somehow instructed the test runner to mark this test as flaky.

Let me see if I can get rid of it though 👍

Reply:
Was able to remove.